plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -160,25 +164,51 @@ func (d *defaultServicePlan) Validate() error {
 	if d.scClient == nil {
 		return errors.New("missing service class interface")
 	}
+	if d.spClient == nil {
+		return errors.New("missing serviceplan interface")
+	}
 	if d.spLister == nil {
 		return errors.New("missing service plan lister")
 	}
 	return nil
 }
 
-func (d *defaultServicePlan) getServiceClassByExternalName(a admission.Attributes, scName string) (*servicecatalog.ClusterServiceClass, error) {
-	glog.V(4).Infof("Fetching serviceclass as %q", scName)
-	listOpts := apimachineryv1.ListOptions{FieldSelector: "spec.externalName==" + scName}
-	servicePlans, err := d.scClient.List(listOpts)
+func (d *defaultServicePlan) getClusterServiceClassByExternalName(a admission.Attributes, scName string) (*servicecatalog.ClusterServiceClass, error) {
+	glog.V(4).Infof("Fetching serviceclass filtered by class external name %q", scName)
+	fieldSet := fields.Set{
+		"spec.externalName": scName,
+	}
+	fieldSelector := fields.SelectorFromSet(fieldSet).String()
+	listOpts := apimachineryv1.ListOptions{FieldSelector: fieldSelector}
+	serviceClasses, err := d.scClient.List(listOpts)
 	if err != nil {
 		glog.V(4).Infof("List failed %q", err)
 		return nil, err
 	}
-	if len(servicePlans.Items) == 1 {
-		glog.V(4).Infof("Found Single item as %+v", servicePlans.Items[0])
-		return &servicePlans.Items[0], nil
+	if len(serviceClasses.Items) == 1 {
+		glog.V(4).Infof("Found Single item as %+v", serviceClasses.Items[0])
+		return &serviceClasses.Items[0], nil
 	}
-	msg := fmt.Sprintf("Could not find a single ServiceClass with name %q", scName)
+	msg := fmt.Sprintf("Could not find a single ServiceClass with name %q, found %v", scName, len(serviceClasses.Items))
 	glog.V(4).Info(msg)
 	return nil, admission.NewNotFound(a)
 }
+
+// getClusterServicePlansByClusterServiceClassName() returns a list of
+// ServicePlans for the specified service class name
+func (d *defaultServicePlan) getClusterServicePlansByClusterServiceClassName(scName string) ([]servicecatalog.ClusterServicePlan, error) {
+	glog.V(4).Infof("Fetching ClusterServicePlans by class name %q", scName)
+	fieldSet := fields.Set{
```

Comment:
same question here, why not just construct the one liner, I find that cleaner to read since we only have a single entry. Again, just curious on the benefit here.